subsys/net/ip/dhcpv4.c

```diff
@@ -39,6 +39,8 @@ static struct k_work_delayable timeout_work;
 
 static struct net_mgmt_event_callback mgmt4_cb;
 
+static struct in_addr request_server_addr;
```

Comment:
Shoudn't this be a part of `iface->config.dhcpv4`? A global variable approach won't work well in case there are mutliple interfaces running DHCP. Ideally I think we could rely on `server_id` instead, but that'd require to sort out the option processing first.

Reply:
Fixed